File: 271-encode-and-decode-strings/271-encode-and-decode-strings.java
public class Codec {

    // Encodes a list of strings to a single string.
    public String encode(List<String> strs) {
            String encodedStr = "";
    for (String str : strs) {
        encodedStr = encodedStr + str.length() + "#" + str;
    }
    return encodedStr;

    }

    // Decodes a single string to a list of strings.
    public List<String> decode(String s) {
        List<String> list = new ArrayList<>();
        for (int startIndex = 0; startIndex < s.length();) {
            int idx = s.indexOf("#", startIndex);
            int numOfChars = Integer.parseInt(s.substring(startIndex, idx));
            list.add(s.substring(idx+1, idx+1+numOfChars));
            startIndex = idx+1+numOfChars;
        }
    return list;
    }
}

// Your Codec object will be instantiated and called as such:
// Codec codec = new Codec();
// codec.decode(codec.encode(strs));

File: 271-encode-and-decode-strings/271-encode-and-decode-strings.py
class Codec:
    def encode(self, strs: [str]) -> str:
        """Encodes a list of strings to a single string.
        """
        return ''.join(f"{len(ss)}|{ss}"for ss in strs)

    def decode(self, s: str) -> [str]:
        """Decodes a single string to a list of strings.
        """
        ans = []
        i = 0
        print(s)
        while i <len(s):
            x1 = s.find("|", i)
            
            i = x1 + 1 + int(s[i:x1])
            ans.append(s[x1+1:i])
            
        return ans
        
        


# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))